app/javascript/store/modules/preferences/actions.js

```diff
@@ -0,0 +1,40 @@
+import preferencesApi from '../../../api/preferences';
+
+import {
+  LOAD_DEFAULT_PREFERENCES,
+  LOAD_TEAM,
+  LOAD_ORGANIZATION,
+  LOAD_TIMESPAN,
+} from '../../action-types';
+
+import {
+  SET_ORGANIZATION,
+  SET_TEAM,
+  SET_TIMESPAN,
+  SET_FETCHING_DEFAULT_PREFERENCES,
+} from '../../mutation-types';
+
+export default {
+  [LOAD_DEFAULT_PREFERENCES]({ commit }, userId) {
+    commit(SET_FETCHING_DEFAULT_PREFERENCES, true);
+
+    return preferencesApi.getPreferences(userId)
+      .then(data => {
```

Comment:
No soy particularmente fanático del `.then` porque se suele usar como si fuera igual que el `await` y no son lo mismo. _same same but not the same_.
El `await` y el `.then` son **no** bloquentes y asincronos de el resto de los métodos que se están ejecutando pero el `await` a diferencia de del `.then` es bloquiante de la línea del método en ejecución. Es decir la línea de abajo de un `await` solo se ejecutará cuando se resuelva la promesa y **no así el `.then`**.
A gusto personal siempre ocupo `await` porque permite evitar comportamientos imprevistos da más "sincronía" y deja el código menos anidado.

Reply:
+1 al `await` por sobre `.then`, creo que dentro de lo complejo que es JS con esto de las promesas, el `async`/`await` es más _developer friendly_ (lectura y escritura) que el then y el catch.